packages/firebase_storage/firebase_storage/windows/firebase_storage_plugin.cpp

```diff
@@ -0,0 +1,807 @@
+        data_, buffer_size_, &putStringListener, controller_);
+    ::Sleep(1);
+
+    future_result.OnCompletion([this](const Future<Metadata>& data_result) {
+      if (data_result.error() == 0) {
+        flutter::EncodableMap event = flutter::EncodableMap();
+        event[kTaskStateName] =
+            static_cast<int>(PigeonStorageTaskState::success);
+        event[kTaskAppName] = std::string(storage_->app()->name());
+        flutter::EncodableMap snapshot = flutter::EncodableMap();
+        snapshot[kTaskSnapshotPath] = data_result.result()->path();
+        snapshot[kTaskSnapshotTotalBytes] = data_result.result()->size_bytes();
+        snapshot[kTaskSnapshotBytesTransferred] =
+            data_result.result()->size_bytes();
+        if (data_result.result()->custom_metadata() != nullptr) {
+          flutter::EncodableMap custom_meta_map = flutter::EncodableMap();
+          for (const auto& kv : *data_result.result()->custom_metadata()) {
+            custom_meta_map[kv.first] = kv.second;
+          }
+          snapshot[kCustomMetadataName] = custom_meta_map;
+        }
+        event[kTaskSnapshotName] = snapshot;
+
+        events_->Success(event);
+      } else {
+        const Error errorCode = static_cast<const Error>(data_result.error());
+        events_->Error(
+            FirebaseStoragePlugin::GetStorageErrorCode(errorCode),
+            FirebaseStoragePlugin::GetStorageErrorMessage(errorCode));
+      }
+    });
+    return nullptr;
+  }
+
+  std::unique_ptr<flutter::StreamHandlerError<flutter::EncodableValue>>
+  OnCancelInternal(const flutter::EncodableValue* arguments) override {
+    return nullptr;
+  }
+
+ public:
+  Storage* storage_;
+  std::string reference_path_;
+  const void* data_;
+  size_t buffer_size_;
+  Controller* controller_;
+  std::unique_ptr<flutter::EventSink<flutter::EncodableValue>>&& events_ =
+      nullptr;
+};
+
+class PutFileStreamHandler
+    : public flutter::StreamHandler<flutter::EncodableValue> {
+ public:
+  PutFileStreamHandler(Storage* storage, std::string reference_path,
+                       std::string file_path, Controller* controller) {
+    storage_ = storage;
+    reference_path_ = reference_path;
+    file_path_ = file_path;
+    controller_ = controller;
+  }
+
+  std::unique_ptr<flutter::StreamHandlerError<flutter::EncodableValue>>
+  OnListenInternal(
+      const flutter::EncodableValue* arguments,
+      std::unique_ptr<flutter::EventSink<flutter::EncodableValue>>&& events)
+      override {
+    events_ = std::move(events);
+
+    TaskStateListener putStringListener = TaskStateListener(events_.get());
+    StorageReference reference = storage_->GetReference(reference_path_);
+    Future<Metadata> future_result =
+        reference.PutFile(file_path_.c_str(), &putStringListener, controller_);
+
+    ::Sleep(1);
+    future_result.OnCompletion([this](const Future<Metadata>& data_result) {
+      if (data_result.error() == 0) {
```

Comment:
In Firestore we had the 
`completed_future.error() == firebase::firestore::kErrorOk`
to check if we had no errors, I wonder if it's a bit more readable

Reply:
done